AdminWebsite/AdminWebsite/ClientApp/src/app/shared/footer/footer.component.ts

```diff
@@ -1,18 +1,33 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { NavigationEnd, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
 import { filter } from 'rxjs/operators';
+import { VersionService } from 'src/app/services/version.service';
 
 @Component({
     selector: 'app-footer',
     templateUrl: './footer.component.html'
 })
-export class FooterComponent implements OnInit {
+export class FooterComponent implements OnInit, OnDestroy {
     hideContactUsLink = false;
+    sub!: Subscription;
+    //private versionService = inject(VersionService);
 
-    constructor(private readonly router: Router) {
+    appVersion: string;
+
+    constructor(
+        private readonly router: Router,
+        private readonly versionService: VersionService
+    ) {
         this.router.events.pipe(filter(event => event instanceof NavigationEnd)).subscribe(x => {
             this.hideContactUs();
         });
+        this.sub = this.versionService.version$.subscribe(version => {
```

Comment:
We've got examples of using the takeUntil operator. Could you please follow the same style.

Reply:
done